ring_buffer/ring_buffer.py

```diff
@@ -1,20 +1,43 @@
 from doubly_linked_list import DoublyLinkedList
+from doubly_linked_list import ListNode
 
 
 class RingBuffer:
     def __init__(self, capacity):
         self.capacity = capacity
         self.current = None
         self.storage = DoublyLinkedList()
+        self.length = 0
+
+
 
     def append(self, item):
```

Comment:
You know I like this code; just for your edification, here's a pretty clean append implementation:
`def append(self, item):`
`        if self.storage.length < self.capacity:`
`            self.storage.add_to_tail(item)`
`            self.current = self.storage.tail`
`        if self.storage.length == self.capacity:`
`            self.current.value = item`
`            if self.current is self.storage.tail:`
`                self.current = self.storage.head`
`            else:`
`                self.current = self.current.next`
So rather than adding and deleting nodes, it's possible to just update the value (assuming the current pointer is in the right place) and then just keep current moving around the ring.